File: car-racing-tournament-web/src/app/models/result.ts
import { Driver } from "./driver";
import { Race } from "./race";
import { ResultType } from "./result-type";
import { Team } from "./team";

export class Result {
  constructor(
    public id: string,
    public type: ResultType,
    public position: number,
    public point: number,
    public driver: Driver,
    public team: Team,
    public race: Race
  ) {}
}

File: car-racing-tournament-api/Models/Result.cs
﻿using System.Text.Json.Serialization;

namespace car_racing_tournament_api.Models
{
    public class Result
    {
        public Guid Id { get; set; }
        public int Position { get; set; }
        public int Point { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Driver Driver { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Team Team { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Race Race { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public Guid DriverId { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public Guid TeamId { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public Guid RaceId { get; set; }
    }
}
